docs/debug/1_getting_started.ipynb

```diff
@@ -0,0 +1,273 @@
+    "The user defines in `config.yaml` which features need to be used in which layers, and then Nvidia-DL-Framework-Inspect takes care of the rest. There are 2 kinds of features:\n",
+    "\n",
+    "- provided by the Transformer Engine - for example, DisableFP8GEMM or LogTensorStats - they are listed in the [debug features API](./3_api_features.ipynb) section\n",
+    "- defined by the user - for example, for testing new precisions - please read the [calls to nvidia-dlframework-inspect](./3_api_te_calls.ipynb) section.\n",
+    "\n",
+    "<figure align=\"center\">\n",
+    "<img src=\"./img/introduction.svg\">\n",
+    "    <figcaption> Fig 1: Example of Nvidia-DL-Framework-Inspect affecting training script with 3 TE Linear Layers. \n",
+    "    There is a specification in `config.yaml` for each layer which features should be used. There are feature class files - some are provided by the TE,\n",
+    "    one - `UserProvidedPrecision` - is implemented by the user. Nvidia-DL-Framework-Inspect inserts features into the layers as it is described in the config.\n",
+    "     </figcaption>\n",
+    "</figure>\n",
+    "\n",
+    "#### Example training script\n",
+    "\n",
+    "Let's look at a simple example of training a Transformer layer using Transformer Engine with FP8 precision. This example demonstrates how to set up the layer, define an optimizer, and perform a few training iterations using dummy data.\n",
+    "\n",
+    "```python\n",
+    "# train.py\n",
+    "\n",
+    "from transformer_engine.pytorch import TransformerLayer\n",
+    "import torch\n",
+    "import torch.nn as nn\n",
+    "import torch.optim as optim\n",
+    "import transformer_engine.pytorch as te\n",
+    "\n",
+    "hidden_size = 512\n",
+    "num_attention_heads = 8\n",
+    "\n",
+    "transformer_layer = TransformerLayer(\n",
+    "    hidden_size=hidden_size,\n",
+    "    ffn_hidden_size=hidden_size,\n",
+    "    num_attention_heads=num_attention_heads\n",
+    ").cuda()\n",
+    "\n",
+    "dummy_input = torch.randn(10, 32, hidden_size).cuda()\n",
+    "criterion = nn.MSELoss()\n",
+    "optimizer = optim.Adam(transformer_layer.parameters(), lr=1e-4)\n",
+    "dummy_target = torch.randn(10, 32, hidden_size).cuda()\n",
+    "\n",
+    "for epoch in range(5):\n",
+    "    transformer_layer.train()\n",
+    "    optimizer.zero_grad()\n",
+    "    with te.fp8_autocast(enabled=True):\n",
+    "        output = transformer_layer(dummy_input)\n",
+    "    loss = criterion(output, dummy_target)\n",
+    "    loss.backward()\n",
+    "    optimizer.step()\n",
+    "```\n",
+    "\n",
+    "We will demonstrate two debug features on the code above:\n",
+    "\n",
+    "1. Disabling FP8 precision for specific GEMM operations, such as the FC1 and FC2 forward propagation GEMM.\n",
+    "2. Logging statistics for other GEMM operations, such as gradient statistics for dgrad GEMM within the LayerNormLinear layer.\n",
```

Comment:
```suggestion
    "2. Logging statistics for other GEMM operations, such as gradient statistics for data gradient GEMM within the LayerNormLinear layer.\n",
```
Before this you should mention something that LayerNormLinear is inside TransformerLayer to give people some context, otherwise this is the first time people hear about it.

Reply:
I meant that they would be hearing about it for the first time in this context - they don't need to understand that TransformerLayer container LayerNormLinear inside it. So we do not need to explain what LayerNormLinear is, but we should mention that TransformerLayer contains it.